scripts/generate_index_pages.ts

```diff
@@ -0,0 +1,161 @@
+import { STATES } from '../src/common';
+
+// We don't care about the values here, but this is a cheap way to determine all
+// of the counties we have any data for and are therefore share-able.
+import CalculatedCountyInterventionJSON from '../src/assets/data/calculated_county_interventions.json';
+const COUNTIES = Object.keys(CalculatedCountyInterventionJSON);
+
+type MetaTags = {[name: string]: string}
+
+function homePageTags(imageUrl: string): MetaTags {
+  // Most of the tags are set correctly in public/index.html so we don't need to
+  // replace them.
+  return {
+    'og:image:url': imageUrl,
+    'twitter:image': imageUrl,
+  };
+}
+
+function locationPageTags(fullImageUrl: string, canonicalUrl: string, locationName: string): MetaTags {
+  const title = `Is ${locationName} ready to reopen?`;
+  const description = 'Real-time modeling and metrics to understand where we stand against COVID.';
+  return {
+    'og:url': canonicalUrl,
+    'og:image:url': fullImageUrl,
+    'og:title': title,
+    'og:description': description,
+
+    'twitter:image': fullImageUrl,
+    'twitter:title': title,
+    'twitter:description': description,
+  };
+}
+
+async function main() {
+  console.log('Building index.html pages...');
+
+  const builder = await IndexPageBuilder.initialize();
+
+  await builder.writeTemplatedPage('/index.html', homePageTags(builder.fullImageUrl('home.png')));
+
+  for (const stateCode in STATES) {
+    const stateName = (STATES as any)[stateCode];
+    const relativeUrl = `/us/${stateCode.toLowerCase()}/`;
+    const canonicalUrl = urlJoin('https://covidactnow.org/', relativeUrl);
+    const imageUrl = builder.fullImageUrl(`states/${stateCode.toLowerCase()}.png`);
+    const page = path.join(relativeUrl, 'index.html');
+    await builder.writeTemplatedPage(page, locationPageTags(imageUrl, canonicalUrl, stateName));
+  }
+
+  for (const fips of COUNTIES) {
+    const county = findCountyByFips(fips);
+    if (!county) {
+      console.warn('Skipping missing county: ', fips);
+      continue;
+    }
+    const stateCode = county.state_code;
+    const locationName = `${county.county}, ${(STATES as any)[stateCode]}`;
+    const relativeUrl = `/us/${stateCode.toLowerCase()}/county/${county.county_url_name}`;
+    const canonicalUrl = urlJoin('https://covidactnow.org/', relativeUrl);
+    const imageUrl = builder.fullImageUrl(`counties/${fips}.png`);
+    const page = path.join(relativeUrl, 'index.html');
+    await builder.writeTemplatedPage(page, locationPageTags(imageUrl, canonicalUrl, locationName));
+  }
+}
+
+class IndexPageBuilder {
+  // TODO(michael): I know parsing HTML via regex is super fraught (could be
+  // fragile) but I'm also hesitant to parse / emit modified HTML. This feels
+  // safer, at least for now.
+  private readonly META_TAG_REGEX =
+      new RegExp(/<meta\s+(?<propKey>(?:property|name))="(?<name>[^"]*?)"\s+content="(?<value>[^"]*?)"/, 'g');
```

Comment:
Thanks! Yeah, that's probably a really good idea. I think it'll mean we'll potentially see the template placeholders (`{{canonicalUrl}}` etc.) when we're debugging locally with `yarn start`, but that shouldn't matter. I doubt anybody is looking at the meta tags anyway.

Since the existing code is working, I may merge as-is. But I'll take a look at swapping in handlebars soon (probably tomorrow). 😁